### ISSUE TITLE
Basic Content Negotiation

### ISSUE THREAD

**Issue by mvriel** [FriendsOfSymfony/FOSRestBundle#329]
Dear maintainers,

For a Restful API that we are working on we would like to implement a basic form of content negotiation using the mimetype. What we would want is the version for JmsSerializer and the Format for JmsSerializer is read from the mimetype.

Example:

```
application/vnd.vendorname.api.user+xml;version=1.0
```

I have figured out how Symfony and FosRestBundle acknowledge the mime type using the mime-type-listener and format_listener in the `app/config/config.yml`:

```
    format_listener:
        default_priorities: ['user', 'xml']
        fallback_format:    'xml'
    view:
        mime_types:
            user: ['application/vnd.vendorname.api.user', 'application/vnd.vendorname.api.user+xml']
```

But what happens now is that the 'user' identifier is passed to JmsSerializer in the Viewhandler's `createResponse` method (https://github.com/FriendsOfSymfony/FOSRestBundle/blob/master/View/ViewHandler.php#L332) which it does not recognize.

Since I just want to use the XML serialization I would love to be able to have JmsSerializer just use the XML format. Unfortunately it seems that the `createResponse` method not only serializes the data using the discovered id but also sets the content type using that same id. This means I cannot separate the serialization format from the content type. 

I have also tried to add a custom Serialization and Deserialization service using the XML Serializer and Deserializer but this poses a lot of other issues (https://github.com/schmittjoh/JMSSerializerBundle/issues/241) where Johannes adviced me to just use the XML serializer instead of trying to add my own.

In short, my question is:

_What would be the best way to separate serialization from content-type discovery so that I can serialize using XML but have all the other handling of `createResponse` use `application/vnd.vendorname.api.user+xml`_

I don't mind to write a PR, but I would like to know your opinion on the above
